docs/design/contextual-tabs.md

```diff
@@ -0,0 +1,392 @@
+
+The contextual tab is registered with Office by calling the [Office.ribbon.requestCreateControls](/javascript/api/office/office.ribbon?view=common-js&preserve-view=true#requestcreateconrtols-input-) method in either the function that is assigned to `Office.initialize` or with the `Office.onReady` method. For more about these methods and initializing the add-in, see [Initialize your Office Add-in](../develop/initialize-add-in.md).
+
+The following is an example. Note that the JSON string must be converted to a JavaScript object with the `JSON.parse` method before it can be passed to a JavaScript function.
+
+```javascript
+Office.onReady(async () => {
+    const contextualTabJSON = ' ... ' // Assign the JSON string such as the one at the end of the preceding section.
+    const contextualTab = JSON.parse(contextualTabJSON);
+    await Office.ribbon.requestCreateControls(contextualTab);
+});
+```
+
+## Specify the contexts when the tab will be visible with requestUpdate
+
+Typically, a contextual tab should appear when a user-initiated event changes the add-in context. Consider a scenario in which the tab should be visible when, and only when, a chart (on the default worksheet of an Excel workbook) is activated.
+
+Begin by assigning handlers. This is commonly done in the **Office.onReady** method as in the following example which assigns handlers (created in a later step) to the **onActivated** and **onDeactivated** events of all the charts in the worksheet.
+
+```javascript
+Office.onReady(async () => {
+    const contextualTabJSON = ' ... ' // Assign the JSON string.
+    const contextualTab = JSON.parse(contextualTabJSON);
+    await Office.ribbon.requestCreateControls(contextualTab);
+
+    await Excel.run(context => {
+        var charts = context.workbook.worksheets
+            .getActiveWorksheet()
+            .charts;
+        charts.onActivated.add(showDataTab);
+        charts.onDeactivated.add(hideDataTab);
+        return context.sync();
+    });
+});
+```
+
+Next, define the handlers. The following is a simple example of a `showDataTab`, but see [Error Handling](#error-handling) below for a more robust version of the function. About this code, note:
+
+- Office controls when it updates the state of the ribbon. The  [Office.ribbon.requestUpdate](/javascript/api/office/office.ribbon?view=common-js&preserve-view=true#requestupdate-input-) method queues a request to update. The method will resolve the Promise object as soon as it has queued the request, not when the ribbon actually updates.
+- The parameter for the **requestUpdate** method is a [RibbonUpdaterData](/javascript/api/office/office.ribbonupdaterdata) object that (1) specifies the tab by its ID *exactly as specified in the JSON*; and (2) specifies visibility of the tab.
+- If you have more than one custom contextual tab that should be visible in the same context, you simply add additional tab objects to the `tabs` array.
+
+```javascript
+async function showDataTab() {
```

Comment:
You could add a visible paremeter to showDataTab so it can be called for show or hide the tab.
async function showDataTab(visible) {
    await Office.ribbon.requestUpdate({
        tabs: [
            {
                id: "CtxTab1",
                visibility: visible // true or false
            }
        ]});
}

Reply:
Yes, but turning the showDataTab into a toggle method makes it more complicated to assign it to events. Currently we have the following which is simple and clear:

`charts.onActivated.add(showDataTab);`
`charts.onDeactivated.add(hideDataTab);`